gateway/proxy.go

```diff
@@ -39,6 +59,27 @@ func customDirector(targetURL *url.URL, originalDirector func(*http.Request)) fu
 	}
 }
 
+func customTransport(component string, timeouts Timeouts) *http.Transport {
+	readTimeout := timeouts.ReadTimeout
+	if readTimeout == 0 {
+		readTimeout = defaultTimeoutValues[component].ReadTimeout
+	}
+	writeTimeout := timeouts.WriteTimeout
+	if writeTimeout == 0 {
+		writeTimeout = defaultTimeoutValues[component].WriteTimeout
+	}
+	idleTimeout := timeouts.IdleTimeout
+	if idleTimeout == 0 {
+		idleTimeout = defaultTimeoutValues[component].IdleTimeout
+	}
+
+	return &http.Transport{
+		ResponseHeaderTimeout: readTimeout,
+		ExpectContinueTimeout: writeTimeout,
+		IdleConnTimeout:       idleTimeout,
+	}
```

Comment:
```suggestion
	t := http.DefaultTransport.(*http.Transport).Clone()
	t.ResponseHeaderTimeout = readTimeout
	// add more timeouts
	return t
```
Transport != DefaultTransport. see https://github.com/golang/go/blob/go1.20.3/src/net/http/transport.go#L38-L54

Reply:
Good to know but I wonder if using a global variable is a good idea. It may lead to bugs that are hard to trace.

If we are going to use the same config as `DefaultTransport`, I propose creating our own `Transport` struct inside `NewProxy` function. What do you think?

Reply:
what do you mean? t is not a global variable and it's calling `Clone()` or am I missing something?

Reply:
Oh, no I missed something 😅. I focused too much on `DefaultTransport` and missed `Clone()`.